httpx/private_ip_validator.go

```diff
@@ -122,6 +135,9 @@ var NoInternalDialer = &net.Dialer{
 	},
 }
 
+// NoInternalTransport
+//
+// DEPRECATED: do not use
```

Comment:
Why deprecate? If someone explicitly chooses to use this, there is no problem right?

Reply:
That's true. However, this transport doesn't allow roundtripper wrapping which is why it is essentially broken / unusable. That's why I decided to deprecate it, because using it most likely means that there is a bug (like in the original PR).